packages/beeq/src/components/tag/bq-tag.tsx

```diff
@@ -0,0 +1,233 @@
+
+  /** Handler to be called when the tag key is pressed */
+  @Event() bqKeyDown: EventEmitter<KeyboardEvent>;
+
+  // Component lifecycle events
+  // Ordered by their natural call order
+  // =====================================
+
+  componentWillLoad() {
+    this.checkPropValues();
+  }
+
+  // Listeners
+  // ==============
+
+  // Public methods API
+  // These methods are exposed on the host element.
+  // Always use two lines.
+  // Public Methods must be async.
+  // Requires JSDocs for public API documentation.
+  // ===============================================
+
+  /** Method to be called to remove the tag component */
+  @Method()
+  async hide(): Promise<void> {
+    this.handleHide();
+  }
+
+  /** Method to be called to show the alert component */
+  @Method()
+  async show(): Promise<void> {
+    this.handleShow();
+  }
+
+  // Local methods
+  // Internal business logic.
+  // These methods cannot be called from the host element.
+  // =======================================================
+
+  private handleHide = () => {
+    const ev = this.bqHide.emit(this.el);
+    if (!ev.defaultPrevented) {
+      this.open = false;
+    }
+  };
+
+  private handleShow = () => {
+    const ev = this.bqShow.emit(this.el);
+    if (!ev.defaultPrevented) {
+      this.open = true;
+    }
+  };
+
+  private handleClick = (event: Event) => {
+    if (this.disabled) {
+      event.preventDefault();
+      event.stopPropagation();
+      return;
+    }
+
+    const ev = this.bqClick.emit(this.el);
+    if (!ev.defaultPrevented) {
+      this.selected = !this.selected;
+    }
+  };
+
+  private handleFocus = (event: Event) => {
+    if (this.disabled) {
+      event.preventDefault();
+      event.stopPropagation();
+      return;
+    }
+
+    this.bqFocus.emit(this.el);
+  };
+
+  private get iconSize(): number {
+    return SIZE_TO_VALUE_MAP[this.size] || SIZE_TO_VALUE_MAP.medium;
+  }
+
+  private handleOnKeyDown = (event: KeyboardEvent) => {
+    this.bqKeyDown.emit(event);
+  };
+
+  private handleSlotChange = () => {
+    this.hasPrefix = hasSlotContent(this.prefixElem, 'prefix');
+  };
+
+  // render() function
+  // Always the last one in the class.
+  // ===================================
+
+  render() {
+    return (
+      <Host
+        class={{ 'is-hidden': !this.open }}
+        aria-hidden={!this.open ? 'true' : 'false'}
+        hidden={!this.open ? 'true' : 'false'}
+      >
+        <div
+          class={{
+            'bq-tag': true,
+            disabled: this.disabled,
+            active: !this.disabled && this.selected,
+            [`bq-tag__wrapper--${this.size} font-medium leading-regular`]: true,
+            [`bq-tag__${this.type}__${this.variant}`]: this.hasColor,
+          }}
+          onClick={this.handleClick}
+          onFocus={this.handleFocus}
+          onKeyDown={this.handleOnKeyDown}
+          role="button"
+          part="wrapper"
+          tabindex={this.disabled ? '-1' : '0'}
+        >
```

Comment:
I think that by changing to `<button>` instead of `<div>`, we can remove the role attribute. The only think is to remove borders on the `.bq-tag` CSS class:

```suggestion
        <button
          class={{
            'bq-tag': true,
            disabled: this.disabled,
            active: !this.disabled && this.selected,
            [`bq-tag__wrapper--${this.size} font-medium leading-regular`]: true,
            [`bq-tag__${this.type}__${this.variant}`]: this.hasColor,
          }}
          onClick={this.handleClick}
          onFocus={this.handleFocus}
          onKeyDown={this.handleOnKeyDown}
          part="wrapper"
          tabindex={this.disabled ? '-1' : '0'}
        >
```

```css
.bq-tag {
  @apply inline-flex border-0 cursor-pointer flex-row items-center justify-center bg-ui-secondary;
```